plugins/commands/compare/src/main/kotlin/CompareCommand.kt

```diff
@@ -167,12 +186,142 @@ class CompareCommand : OrtCommand(
                     throw ProgramResult(0)
                 }
 
+                val diff = resultA.diff(resultB)
+
+                echo(deserializer.writeValueAsString(diff))
+
                 throw ProgramResult(1)
             }
         }
     }
 }
 
+private fun OrtResult.diff(other: OrtResult) =
+    OrtResultDiff(
+        repositoryDiff = repository.diff(other.repository)
+    )
+
+private fun Repository.diff(other: Repository): RepositoryDiff? =
```

Comment:
Hmm, the following code looks the laborious. I wonder whether we could make some smart re-use of Kotest matcher code here?

Reply:
I also thought about using this, but I guess the output of those Kotest matcher would be not really readable for large files/big diffs...